### PR TITLE
more coverity fixes

### PR DESCRIPTION
**Pull request by h4ng3r**

### CODE REVIEW

File: libr/core/cmd_print.c [hunk 1]
```diff
@@ -1629,6 +1629,7 @@ static void disasm_strings(RCore *core, const char *input, RAnalFunction *fcn) {
 	free (string2);
 	free (string);
 	free (s);
+	free (str);
```
Comment (alvarofe) [inline libr/core/cmd_print.c]: does this string always contain allocated memory?

Comment (h4ng3r) [inline libr/core/cmd_print.c]: well, coverity complains but maybe you are right and it is not always allocated.
Maybe we can free after "flag = r_flag_get_i2 (core->flags, ptr);". Seems better to you?

Comment (alvarofe) [inline libr/core/cmd_print.c]: mm i don't see why after that, the thing is that str is only allocated at (at least i am aware of)

```
		if (str) {
			char *qoe = strstr (str, ";");
			if (qoe) {
				//XXX str leaks
				str = r_str_ndup (str, qoe - str);
			}
		}

```

but the rest of cases is just a pointer returned by strstr

Comment (h4ng3r) [inline libr/core/cmd_print.c]: true, but where should be place the free then?

**Comment by h4ng3r** [comment]
I removed the R_FREE (str) and moved free (name)

File: libr/core/cmd_print.c (hunk 1 above)
Comment (alvarofe) [inline libr/core/cmd_print.c]: what about a variable to track when free it. Afterwards run some test with asan or valgrind such as t.esil/apple iirc that path is triggered.

Comment (h4ng3r) [inline libr/core/cmd_print.c]: done ;)

File: libr/core/cmd_print.c [hunk 2]
```diff
@@ -1484,6 +1485,7 @@ static void disasm_strings(RCore *core, const char *input, RAnalFunction *fcn) {
 				int len = qoe - qo - 1;
 				if (len > 2) {
 					string = r_str_ndup (qo, len+2);
+					is_free_pending = true;
```
Comment (alvarofe) [inline libr/core/cmd_print.c]: `str` was leaking not string. It should be here

```
		if (str) {
			char *qoe = strstr (str, ";");
			if (qoe) {
				//XXX str leaks
				str = r_str_ndup (str, qoe - str);
			}
		}

```

Comment (h4ng3r) [inline libr/core/cmd_print.c]: yep, of course. I dont know why I placed it there :(

File: libr/core/cmd_print.c [hunk 3]
```diff
@@ -1629,6 +1631,9 @@ static void disasm_strings(RCore *core, const char *input, RAnalFunction *fcn) {
 	free (string2);
 	free (string);
 	free (s);
+	if (is_free_pending) {
+		free(str);
```
Comment (radare) [inline libr/core/cmd_print.c]: missing space before (

Comment (h4ng3r) [inline libr/core/cmd_print.c]: done, thx for the comment.

**Pull request merged**